### PR TITLE
updated fprime types for correct compilation with vxworks and baremetal

### PR DESCRIPTION
**Pull request by kevin-f-ortega**
| | |
|:---|:---|
|**_Originating Project/Creator_**|@kevin-f-ortega |
|**_Affected Component_**|  |
|**_Affected Architectures(s)_**|  |
|**_Related Issue(s)_**|  |
|**_Has Unit Tests (y/n)_**|  |
|**_Builds Without Errors (y/n)_**|  y|
|**_Unit Tests Pass (y/n)_**|  |
|**_Documentation Included (y/n)_**|  |

---
## Change Description

Fixed bugs founds when compiling for baremetal,  added the PLATFORM_*_TYPE macros, updated RateGroupDriver to use the configure pattern.

## Rationale

Bug fixes.
To be more consistent.

## Testing/Review Recommendations

PLATFORM_*_TYPES should be reviewed

### CODE REVIEW

File: Svc/RateGroupDriver/RateGroupDriverImpl.hpp [hunk 1]
```diff
@@ -40,17 +40,21 @@ namespace Svc {
             //!  for use when the CycleIn port is called.
             //!
             //!  \param compName component name
-            //!  \param dividers array of integers used to divide down input tick
-            //!  \param numDividers size of dividers array
             //!
             //!  \return return value description
-            RateGroupDriverImpl(const char* compName, NATIVE_INT_TYPE dividers[], NATIVE_INT_TYPE numDividers);
```
Comment (bocchino) [inline Svc/RateGroupDriver/RateGroupDriverImpl.hpp:47]: This change requires updates to the Ref topology and the RPI topology.

File: Fw/Types/DefaultTypes.hpp [hunk 2]
```diff
@@ -0,0 +1,15 @@
+
+#ifndef PLATFORM_INT_TYPE_DEFINED
+typedef int PLATFORM_INT_TYPE;
+#define PRIfwpit "d"
```
Comment (bocchino) [inline Fw/Types/DefaultTypes.hpp:4]: Shouldn't this be "%d"?

File: Fw/Types/DefaultTypes.hpp [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+#endif
+#ifndef PLATFORM_UINT_TYPE_DEFINED
+typedef unsigned int PLATFORM_UINT_TYPE;
+#define PRIfwpuit "u"
```
Comment (bocchino) [inline Fw/Types/DefaultTypes.hpp:8]: Shouldn't this be "%u"?

**Comment by bocchino** [comment]
Before merging this PR, we should add some documentation saying what the new types are and how to use them. Also, we should add the new types as known types in the FPP tools.

File: Fw/Types/DefaultTypes.hpp (hunk 2 above)
Comment (kevin-f-ortega) [reply Fw/Types/DefaultTypes.hpp:4]: No. In C/C++, the PRI* macros are defined without the "%" symbol.

**Comment by LeStarch** [comment]
I am being blocked from commenting for some reason so:

Do we want FW_INDEX_TYPE to read FwIndex or FwIndexType. NATIVE_INT_TYPE was a #define and thus used #define capitalization. We have formalized FW index into a type, so should it use type naming like AssertArg?

**Comment by bocchino** [comment]
> We have formalized FW index into a type, so should it use type naming like AssertArg

Yes, type names should be in CapitalCase, not CAPITAL_SNAKE_CASE. Good catch.

File: Svc/RateGroupDriver/RateGroupDriverImpl.cpp [hunk 4]
```diff
@@ -6,47 +6,50 @@
...
+
+    RateGroupDriverImpl::~RateGroupDriverImpl() {
+
+    }
+
+    void RateGroupDriverImpl::init(FW_INDEX_TYPE instanceId) {
+        RateGroupDriverComponentBase::init(instanceId);
+    }
 
+    void RateGroupDriverImpl:: configure(const FW_INDEX_TYPE dividers[], const FW_SIZE_TYPE numDividers) {
```
Comment (LeStarch) [inline Svc/RateGroupDriver/RateGroupDriverImpl.cpp:24]: Add assert to prevent failure to call configure.

**Comment by LeStarch** [comment]
@kevin-f-ortega can we close this PR?

**Comment by kevin-f-ortega** [comment]
To be broken up into smaller PRs.

**Pull request closed without merging**